services/checks/checks.go

```diff
@@ -419,6 +420,118 @@ func (s *Service) EnableChecks(checkNames []string) error {
 	return nil
 }
 
+// ChangeInterval changes a check's interval to the value received from the UI.
+func (s *Service) ChangeInterval(params map[string]managementpb.SecurityCheckInterval) error {
+	checks := s.GetAllChecks()
+	if len(checks) == 0 {
+		return errors.New("no checks loaded")
+	}
+
+	checkMap := make(map[string]check.Check, len(checks))
+	for _, ch := range checks {
+		checkMap[ch.Name] = ch
+	}
+
+	checkStateMap, err := models.FindCheckStates(s.db.Querier)
+	if err != reform.ErrNoRows && err != nil {
+		return errors.Wrap(err, "failed to retrieve checks state: %+v")
+	}
+
+	for name, interval := range params {
+		c, ok := checkMap[name]
+		if !ok {
+			return errors.Errorf("check: %s not found", name)
+		}
+
+		switch interval {
+		case managementpb.SecurityCheckInterval_STANDARD:
+			c.Interval = check.Standard
+		case managementpb.SecurityCheckInterval_FREQUENT:
+			c.Interval = check.Frequent
+		case managementpb.SecurityCheckInterval_RARE:
+			c.Interval = check.Rare
+		case managementpb.SecurityCheckInterval_SECURITY_CHECK_INTERVAL_INVALID:
+			return errors.New("invalid security check interval")
+		default:
+			return errors.New("unknown security check interval")
+		}
+
+		s.updateCheck(c)
+
+		// since we re-run checks at regular intervals using a call
+		// to s.StartChecks which in turn calls s.collectChecks
+		// to load/download checks, we must persist any changes
+		// to check intervals in the DB so that they can be re-applied
+		// once the checks have been re-loaded on restarts.
+		if _, ok := checkStateMap[c.Name]; !ok {
+			cs, err := models.CreateCheckState(s.db.Querier, c.Name, models.Interval(c.Interval))
+			if err != nil {
+				return err
+			}
+			s.l.Debugf("Saved interval change for check: %s in DB", cs.Name)
+			continue
+		}
```

Comment:
It can fail because you grab existing without transaction, so some other user can create a check state. So I think all DB interaction should be wrapped into the transaction.